packages/compass-sidebar/src/components/sidebar-instance/sidebar-instance.jsx

```diff
@@ -1,94 +1,79 @@
+import FavoriteButton from '../favorite-button';
 
 import styles from './sidebar-instance.module.less';
 
-class SidebarInstance extends PureComponent {
-  static displayName = 'SidebarInstance';
-  static propTypes = {
-    instance: PropTypes.object,
-    databases: PropTypes.array,
-    isExpanded: PropTypes.bool.isRequired,
-    isGenuineMongoDB: PropTypes.bool.isRequired,
-    toggleIsDetailsExpanded: PropTypes.func.isRequired,
-    globalAppRegistryEmit: PropTypes.func.isRequired,
-    detailsPlugins: PropTypes.array.isRequired,
-    connectionModel: PropTypes.object,
-    toggleIsModalVisible: PropTypes.func.isRequired,
-    isModalVisible: PropTypes.bool.isRequired,
-    saveFavorite: PropTypes.func.isRequired
-  };
+export const SidebarInstance = ({
+  instance,
+  databases,
+  isExpanded,
+  isGenuineMongoDB,
+  toggleIsDetailsExpanded,
+  globalAppRegistryEmit,
+  detailsPlugins,
+  connectionInfo,
+  updateConnectionInfo
+}) => {
+  const [ isFavoriteModalVisible, setIsFavoriteModalVisible ] = useState(false);
 
-  /**
-   * Closes the favorite modal.
-   */
-  closeFavoriteModal() {
-    this.props.toggleIsModalVisible(false);
-  }
+  const onClickSaveFavorite = useCallback((newFavoriteInfo) => {
+    updateConnectionInfo({
+      ...connectionInfo,
+      favorite: newFavoriteInfo
+    });
 
-  /**
-   * Saves the current connection to favorites.
-   *
-   * @param {String} name - The favorite name.
-   * @param {String} color - The favorite color.
-   */
-  saveFavorite(name, color) {
-    this.props.saveFavorite(this.props.connectionModel.connection, name, color);
-    this.props.toggleIsModalVisible(false);
-  }
+    setIsFavoriteModalVisible(false);
+  }, [connectionInfo, updateConnectionInfo, setIsFavoriteModalVisible]);
 
-  /**
-   * Renders the favorite modal.
-   *
-   * @returns {React.Component}
-   */
-  renderFavoriteModal() {
-    if (this.props.isModalVisible) {
-      return (
-        <FavoriteModal
-          connectionModel={this.props.connectionModel.connection}
-          closeFavoriteModal={this.closeFavoriteModal.bind(this)}
-          saveFavorite={this.saveFavorite.bind(this)}
-        />
-      );
-    }
-  }
+  return (
+    <div className={styles['sidebar-instance']}>
+      <SidebarInstanceStats
+        instance={instance}
+        databases={databases}
+        isExpanded={isExpanded}
+        toggleIsExpanded={toggleIsDetailsExpanded}
+        globalAppRegistryEmit={globalAppRegistryEmit}
+      />
+      <FavoriteButton
+        favoriteOptions={connectionInfo.favorite}
+        toggleIsFavoriteModalVisible={() => setIsFavoriteModalVisible(
+          !isFavoriteModalVisible
+        )}
+      />
+      <SaveConnectionModal
+        initialFavoriteInfo={connectionInfo.favorite}
+        open={isFavoriteModalVisible}
+        onCancelClicked={() => setIsFavoriteModalVisible(false)}
+        onSaveClicked={(favoriteInfo) => onClickSaveFavorite(favoriteInfo)}
+      />
```

Comment:
So this new form is not behind any feature flag and allows user to choose color, which doesn't change anything if you are not using new connection form (and all the other things that come with it), is this expected?

Reply:
Ah, got it, thanks for the context! Yeah, totally if you want to do it that way! FWIW the project seems to be wrapping up pretty soon and it's such a small thing that if you feel like merging it already I think it's totally fine, but I trust your judgement here